pipeline-modules/app-service/gcp/node-build.yaml

```diff
@@ -0,0 +1,56 @@
+provisioner: gcp
+name: node-build
+version: 1
+revision: 1
+displayName: Node.js Build
+description: This plugin is for building or testng your app with Node.js
+target: ""
+category: build
+keywords:
+  - Node.js
+  - NPM
+  - Yarn
+  - Install
+  - Build
+author: CloudCover
+meta: {}
+inputs:
+  properties:
+    package_manager:
+      title: Package manager
+      description: Your package manager, either node or yarn
+      type: string
+      default: node
+      enum:
+        - node
+        - yarn
+    npm_build_args:
+      title: Package manager arguments
+      description: Arguments for your Package manager command
+      type: array
+      default: ["install"]
+      items:
+        type: string
+    working_dir:
+      title: Working directory
+      description: The current working directory to execute command
+      type: string
+      default: repo
+    node_js_version:
+      title: Node.js Version
+      description: The version of Node.js being used (From https://hub.docker.com/_/node/)
+      type: string
+      default: 18.12.0-alpine
+  # For some reason required fields with default value is not allowed, this needs to be fixed before we uncomment the below fields
```

Comment:
I wasn't aware of this - we definitely have other templates that have required fields that are defaulted -what behaviour are you seeing?

Reply:
```panic: could not load pipeline modules: module file '/Volumes/repos/codepipes-ci-cd-modules/pipeline-modules/app-service/gcp/node-build.yaml': required property 'package_manager' must not define default value

goroutine 1 [running]:
github.com/cldcvr/vanguard-api/pkg/pipeline.failModuleLoad(...)
	/Users/nikhilverma/go/pkg/mod/github.com/cldcvr/vanguard-api@v0.24.1-0.20221211154702-8221d7066096/pkg/pipeline/manager.go:138
github.com/cldcvr/vanguard-api/pkg/pipeline.newManager({0x106be27a8, 0x140001a2000}, {0x106be0740?, 0x140004a8d60})
	/Users/nikhilverma/go/pkg/mod/github.com/cldcvr/vanguard-api@v0.24.1-0.20221211154702-8221d7066096/pkg/pipeline/models.go:294 +0x164
github.com/cldcvr/vanguard-api/pkg/pipeline.NewManager(...)
	/Users/nikhilverma/go/pkg/mod/github.com/cldcvr/vanguard-api@v0.24.1-0.20221211154702-8221d7066096/pkg/pipeline/provider.go:22
main.seed({0x105fec9bb, 0xb}, {0x14000054107, 0x37})
	/Volumes/repos/codepipes-ci-cd-modules/cpi-module-seed/main.go:138 +0x2d8
main.cmdSeed()
	/Volumes/repos/codepipes-ci-cd-modules/cpi-module-seed/main.go:106 +0x10c
main.main()
	/Volumes/repos/codepipes-ci-cd-modules/cpi-module-seed/main.go:65 +0x15c
exit status 2
```

Reply:
@NikhilVerma ok - let's move forward with your PR the way it is - separately can you file a PR for this and we can get @petmal to take a look